internal/openssh/lib.go

```diff
@@ -0,0 +1,214 @@
+package openssh
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/binary"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"golang.org/x/crypto/ssh"
+)
+
+// This module cherry-picks from https://go-review.googlesource.com/c/crypto/+/218620,
+// that provides a solution for https://github.com/golang/go/issues/37132, that highlights
+// the need for methods to marshal private keys to the OpenSSH format in `x/crypto/ssh`.
+//
+// This provides utilities to serialize a Private Keys in OpenSSH format: once
+// this goes mainstream, we can remove this module.
+
+// Constants represents the algorithm names for key types supported by this package.
+const (
+	KeyAlgoRSA = "ssh-rsa"
+
+	CurveNIST256 = "nistp256"
+	CurveNIST384 = "nistp384"
+	CurveNIST521 = "nistp521"
+
+	KeyAlgoECDSA256 = "ecdsa-sha2-nistp256"
+	KeyAlgoECDSA384 = "ecdsa-sha2-nistp384"
+	KeyAlgoECDSA521 = "ecdsa-sha2-nistp521"
+
+	KeyAlgoED25519 = "ssh-ed25519"
+)
+
+const magic = "openssh-key-v1\x00"
+
+// MarshalPrivateKey returns a *pem.Block with the private key serialized in the OpenSSH format.
+//
+// See: https://coolaj86.com/articles/the-openssh-private-key-format/
+//
+// NOTE: `x/crypto/ssh` doesn't handle elliptic curve P-224, and so this applies to this function as well.
```

Comment:
nit: I think its actually more fundamental that OpenSSH and the SSH protocol don't handle P-224 either. Outside of X.509, it isn't that widely used curve as P-256 and P-521 are fast enough for 99% of use cases and likely provide additional security.

Reply:
Will rephrase.